### ISSUE TITLE
[Feature]: <Gradient Boosting Machines (GBM)>

### ISSUE THREAD

**Issue by priyashuu** [Ajay-Dhangar/algo#1095]
### Feature Name

Adding Gradient Boosting Machines (GBM) Visualizations

### Feature Description

Gradient Boosting Machines algorithm, which builds models sequentially and corrects errors made by previous models. This will include popular variants like XGBoost and LightGBM.

### Motivation

GBM techniques are widely used in machine learning competitions and real-world applications due to their high performance. Adding this feature will provide users with advanced modeling capabilities.

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [X] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other
